### PR TITLE
test: avoid unconditional reverts in stack traces tests

### PR DESCRIPTION
**Pull request by fvictorio**
Re-applying the changes from https://github.com/NomicFoundation/hardhat/pull/5467

See discussion in https://github.com/NomicFoundation/hardhat/issues/5443

### CODE REVIEW

**Review by Xanewok — APPROVED** [review]

**Review by Wodann — APPROVED** [review]

**Comment by fvictorio** [comment]
Thanks for the reviews. I'm going to merge this after the PR in Hardhat is merged.

**Pull request merged**
